### ISSUE TITLE
Rules should be combined by logical OR

### ISSUE THREAD

**Issue by GVanderLugt** [fratzinger/feathers-casl#17]
The CASL documentation states that rules should be combined by logical `OR`. `feathers-casl` is combining rules by logical `AND`.

I added a test to help demonstrate this.

```typescript
it("should combine rules by logical OR", async function() {
  await service.create({ test: true, userId: 1, published: false });
  await service.create({ test: true, userId: 2, published: true });

  const items = (await service.find({ paginate: false })) as unknown[];
  assert(items.length === 2, "has two items");

  const returnedItems = await service.find({
    //@ts-ignore
    ability: defineAbility(
      (can) => {
        can("read", "tests", { published: true });
        can("read", "tests", { userId: { $in: [1] } });
      },
      { resolveAction }
    ),
    paginate: false,
  });

  // @ts-ignore
  assert.strictEqual(returnedItems.length, 2);
});
```

Combining these rules by logical OR, we should be getting all records where `published=true OR userId IN [1]`. This would result in both created records being returned. Instead, we get no records because the query is where `published=true AND userId IN [1]`.

I'm not exactly sure where to go from here. I tried pushing a branch with this test, but don't have permission. I looked through the `getQueryFor` logic, and it's pretty complex. @fratzinger I see that you're not using [`rulesToQuery`](https://casl.js.org/v5/en/api/casl-ability-extra#rules-to-query) or other helpers that CASL provides for this kind of thing. Is there a reason for that?

**Comment by fratzinger** [comment]
Please fork the repo, do the changes to your fork and make a PR with the new test.
Would you please also add a test with:
```
    can("read", "tests", {
      $or: [
        { published: true }
        { user_id: { $in: [1] } },
      ]
    });
}
```
I know `Casl` states out, that this is not possible, but I would like to see the outcome.

>  I see that you're not using rulesToQuery or other helpers that CASL provides for this kind of thing. Is there a reason for that?
As stated on slack before, the following will not work properly with `rulesToQuery`:

rule: `can('find', 'posts', { id: { $in: [1, 2, 3] } })`
query: `service('posts').find({ query: { id: { $in: [1, 2] } } } );`

**Expected:**
finds 2 items with ids 1 and 2

**Actually:**
finds 3 items with ids 1, 2 and 3

**Comment by fratzinger** [comment]
The problem with `rulesToQuery` is, that not all adapters support `$not`, `$and` and `$nor` (`toMongoQuery`).
I've started a new branch: https://github.com/fratzinger/feathers-casl/tree/multiple-adapters to test against most of the popular adapters.

**Comment by fratzinger** [comment]
Please have a look at: https://github.com/fratzinger/feathers-casl/tree/logical-or that consequently integrates the `$or` logical

**Comment by fratzinger** [comment]
I consider this ready now for `v0.5.0` but I need some investigations. Stay tuned.

**Comment by fratzinger** [comment]
released as [v0.5.0](https://github.com/fratzinger/feathers-casl/releases/tag/v0.5.0)
